src/app/games/mancala/kalah/tests/kalah.component.spec.ts

```diff
@@ -0,0 +1,331 @@
+        it('should allow double distribution move', fakeAsync(async() => {
+            // Given any board where first distribution has been done
+            await mancalaTestUtils.expectMancalaClickSuccess(new Coord(3, 1));
+            // When doing double distribution move
+            const move: KalahMove = KalahMove.of(MancalaDistribution.THREE, [MancalaDistribution.ZERO]);
+            // Then it should be a success
+            await mancalaTestUtils.expectMancalaMoveSuccess('#click_0_1', move);
+        }));
+        it('should hide last move when doing illegal click during complex move', fakeAsync(async() => {
+            // Given a distribution that just ended up in the Kalah
+            await mancalaTestUtils.expectMancalaClickSuccess(new Coord(3, 1));
+
+            // When doing an illegal click
+            const reason: string = MancalaFailure.MUST_CHOOSE_NON_EMPTY_HOUSE();
+            await mancalaTestUtils.testUtils.expectClickFailure('#click_3_1', reason);
+
+            // Then the last move should be hidden
+            mancalaTestUtils.expectHouseToContain(new Coord(3, 1), ' 4 ');
+            mancalaTestUtils.expectHouseToContain(new Coord(2, 1), ' 4 ');
+            mancalaTestUtils.expectHouseToContain(new Coord(1, 1), ' 4 ');
+            mancalaTestUtils.expectHouseToContain(new Coord(0, 1), ' 4 ');
+            mancalaTestUtils.expectStoreContentToBe(Player.ZERO, ' 0 ');
+            mancalaTestUtils.testUtils.expectElementToHaveClasses('#circle_3_1', ['base', 'player0-fill']);
+            // The filled spaces
+            mancalaTestUtils.testUtils.expectElementToHaveClasses('#circle_2_1', ['base', 'player0-fill']);
+            mancalaTestUtils.testUtils.expectElementToHaveClasses('#circle_1_1', ['base', 'player0-fill']);
+            mancalaTestUtils.testUtils.expectElementToHaveClasses('#circle_0_1', ['base', 'player0-fill']);
+            mancalaTestUtils.testUtils.expectElementToHaveClasses('#circle_-1_-1', ['base', 'player0-fill']);
+        }));
+        it('should allow triple distribution move', fakeAsync(async() => {
+            // Given a state where multiple capture are possible
+            const state: MancalaState = new MancalaState([
+                [6, 1, 7, 6, 1, 7],
+                [2, 1, 6, 2, 2, 5],
+            ], 3, [4, 2]);
+            await mancalaTestUtils.testUtils.setupState(state);
+
+            // When doing the complex move
+            await mancalaTestUtils.expectMancalaClickSuccess(new Coord(0, 0));
+            await mancalaTestUtils.expectMancalaClickSuccess(new Coord(4, 0));
+            const move: KalahMove = KalahMove.of(MancalaDistribution.ZERO,
+                                                 [MancalaDistribution.FOUR, MancalaDistribution.ONE]);
+
+            // Then the move should be legal
+            await mancalaTestUtils.expectMancalaMoveSuccess('#click_1_0', move);
+        }));
+        it('should allow triple distribution move (2)', fakeAsync(async() => {
```

Comment:
pourquoi on a besoin de deux tests pour la même chose ? Si on teste exactement la même chose, faut pas deux tests; si on teste pas exactement la même chose, je dirais que le it devrait être légèrement différent